pkg/model-serving-controller/controller/model_serving_controller.go

```diff
@@ -697,6 +701,14 @@ func (c *ModelServingController) DeleteRole(ctx context.Context, mi *workloadv1a
 		return
 	}
 	for _, svc := range services {
+		svcDeepCopy := svc.DeepCopy()
+		// remove finalizer first to avoid being blocked by deleting headless service
+		utils.RemoveHeadlessServiceProtectionFinalizer(svcDeepCopy)
+		_, err = c.kubeClientSet.CoreV1().Services(mi.Namespace).Update(context.TODO(), svcDeepCopy, metav1.UpdateOptions{})
+		if err != nil {
+			klog.Errorf("failed to update service %s: %v to remove finalizer", svcDeepCopy.Name, err)
+			continue
+		}
```

Comment:
I mean you can wrap this as a function with delete, so we donot need to care about finalizers every where we need to delete

Like 
```golang 
func DeleteService(){

 1. remove finalizer
 2. update
 3. delete

}
```